extensions/sql-database-projects/src/common/utils.ts

```diff
@@ -626,3 +626,32 @@ export function getFoldersAlongPath(startFolder: string, endFolder: string): str
 
 	return folders;
 }
+
+/**
+ * Determines if provided value is a well-known database source and therefore is allowed to be sent in telemetry
+ * @param value Value to check if it's a well-known database source
+ * @returns Database source value if well-known, otherwise returns undefined
+ */
+export function getWellKnownDatabaseSourceValue(value: string): string | undefined {
+	const upperCaseValue = value.toUpperCase();
+	return constants.WellKnownDatabaseSources
+		.find(wellKnownSource => wellKnownSource.toUpperCase() === upperCaseValue);
+}
+
+/**
+ * Get string containing all well-known database sources, separated by semicolon
+ * @param databaseSourceValues Array of database source values
+ * @returns Well-known database sources, separated by semicolon
+ */
+export function getWellKnownDatabaseSourceString(databaseSourceValues: string[]): string {
```

Comment:
seems unintuitive that the side-effect of this method is to convert from an array to a semicolon-delimited list

Reply:
I agree with you here.  I think it may be better to simply return string[] and create the semicolon-delimited list later.